ament_clang_tidy/ament_clang_tidy/main.py

```diff
@@ -259,6 +294,76 @@ def get_compilation_db_files(paths):
     return [os.path.normpath(f) for f in files]
 
 
+def filter_packages_in_repo(compilation_dbs, repo_path):
+    bin_names = [
+        'catkin_topological_order',
+    ]
+    catkin_topological_order_bin = find_executable(bin_names)
+    if not catkin_topological_order_bin:
+        print('Could not find %s executable' %
+              ' / '.join(["'%s'" % n for n in bin_names]), file=sys.stderr)
+        return 1
+
+    cmd = [catkin_topological_order_bin, repo_path, '--only-names']
+    output = ''
+    try:
+        output = subprocess.check_output(cmd).strip().decode()
+    except subprocess.CalledProcessError as e:
+        print('The invocation of "%s" failed with error code %d: %s' %
+              (os.path.basename(catkin_topological_order_bin), e.returncode, e),
+              file=sys.stderr)
+
+    ros_packages = output.split()
+    test = lambda path: os.path.basename(os.path.dirname(path)) in ros_packages
+    filtered_dbs = list(filter(test, compilation_dbs))
+    return filtered_dbs
+
+
+def filter_diff(compilation_dbs, repo_path, diff_head):
```

Comment:
Only supporting git and searching for the binary explicitly is too restrictive IMO, using `vcs-branch`/`vcs-diff` removes a lot of logic from here and offers support for other vcs tools. https://github.com/dirk-thomas/vcstool

Reply:
Because of the nature of vcs it has to implement only the features common ammong version control systems.  A second issue is that vcs is intended to be run in a source directory and crawl for repos.  I need to make a query against one specific repo.  As a result of the requirements on vcstool there is no interface in vcs for asking for the diff against a specific branch, filtering the types of diffs, or limiting results to just filenames, or limiting the diff to a directory within a repo.

For context here is a typical git diff command I want to run:
```bash
git diff --name-only --diff-filter=MA master..HEAD <src/repo/package>
```